src/plugins/editor-monaco-language-apidom/language/providers/DiagnosticsProvider.js

```diff
@@ -94,10 +94,12 @@ class DiagnosticsProvider extends Provider {
   async #validate(model) {
     const diagnostics = await this.#getDiagnostics(model);
 
-    this.#diagnosticCollection?.set(
-      model.uri,
-      await this.protocolConverter.asDiagnostics(diagnostics)
-    );
+    if (this.protocolConverter) {
```

Comment:
We can make this more explicit. The `?` character is very subtle an can be overlooked. We need both of these components to execute the code.

```suggestion
    if (this.protocolConverter && this.#diagnosticCollection) {
```